test/test_tensorflow.py

```diff
@@ -253,6 +261,11 @@ def test_horovod_allreduce_error(self):
         rank = hvd.rank()
         size = hvd.size()
 
+        # There is no checking if tensor sizes from different ranks are equal in MLSL.
```

Comment:
Why not use existing mechanism we have in Horovod to ensure shape compatibility?  This way, UX is the same across backends.

Reply:
@alsrgv Thank you for your comment. Unfortunately, we cannot use `ConstructMPIResponse` function as is because in MLSL implementation we don't have and don't need the data types used in this function and would like to avoid extra data copies. The second point is that in our implementation coordinator doesn't know anything about the tensors of other processes (one sided communication). All the other processes can compare their tensors with the one of coordinator, but the decision about error will be made in a decentralized way that can lead to the situation when coordinator is still working with troubled tensor but others are waiting it to go to Finalize phase. We just wouldn't like to add some extra communications between coordinator and the other processes. MPI will report about different sizes and types anyway. And if such kind of situation exist then something should be changed in the way which data collective operations are called with. I unserstand your desire to keep UX the same, but the question is, how important it is to implement error reporting feature inside our piece of code?

Reply:
I'd like to understand this problem deeper.  We should not diverge from approach used by original coordinator implementation to keep code base simple, unless there's potential for significant performance improvements - in which case we may in fact want to implementation of the original code base.  Sent you email to discuss over V/C.  Let's keep this open to summarize our discussion here afterwards.